base/Dockerfile

```diff
@@ -42,19 +42,45 @@ RUN useradd -l -m -u $UID techuser && \
     && chown techuser /home/techuser
 ENV HOME=/home/techuser
 
-
 # This in analogous to the virtualenv activate script
 # To allow deactivation of the virtualenv, we need to save the old PATH
 ENV _OLD_VIRTUAL_PATH="$PATH"
 ENV VIRTUAL_ENV=/opt/.venv
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
+# hadolint ignore=DL3003
 RUN ln -s "$(which python3.11)" /usr/bin/python && \
     ln -sf "$(which python3.11)" /usr/bin/python3 && \
     ln -sf "$(which pip3.11)" /usr/local/bin/pip && \
     ln -sf "$(which pip3.11)" /usr/local/bin/pip3 && \
     python -m venv /opt/.venv && \
     chmod -R 777 /opt/.venv/bin/ && \
-    chmod -R 777 /opt/.venv/lib/python3.11/site-packages
+    chmod -R 777 /opt/.venv/lib/python3.11/site-packages && \
+    # Configure pre-commit
+    pip install pre-commit==3.4.0 --no-cache-dir && \
+    mkdir -p /opt/git/global-hooks && \
+    mkdir /tmp/pre-commit-init && cd /tmp/pre-commit-init && \
+    git init && \
+    rm .git/hooks/* && \
+    pre-commit install \
+        --allow-missing-config \
+        -t commit-msg \
+        -t post-checkout \
+        -t post-commit \
+        -t post-merge \
+        -t post-rewrite \
+        -t pre-commit \
+        -t pre-merge-commit \
+        -t pre-push \
+        -t pre-rebase \
+        -t prepare-commit-msg && \
```

Comment:
It's a pitty there's no way to install pre-commit hooks in a more straightforward way.

Reply:
Had to reimplement due to a deadlock with `pre-commit`. It looks a bit better now.